eth/backend.go

```diff
@@ -89,12 +89,22 @@ func (b *backend) SendTransaction(ctx context.Context, tx *types.Transaction) er
 		method = "unknown"
 	}
 
-	if err != nil {
-		glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, err, strings.Repeat("*", 75))
+	txParams := make(map[string]interface{})
+	if err := decodeTxParams(b.abiMap[string(data[:4])], txParams, data); err != nil {
 		return err
 	}
 
-	glog.Infof("\n%vEth Transaction%v\n\nInvoking transaction: \"%v\".  Hash: \"%v\". \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), method, tx.Hash().String(), strings.Repeat("*", 75))
+	var txParamsString string
+	for name, val := range txParams {
```

Comment:
We can list the tx params in the order that they are included in tx data by relying on the ordering of the `Inputs` slice stored in an ABI. Something like:

```golang
for _, arg := range abi.Methods[method].Inputs {
    txParamsString += fmt.Sprintf("%v: %v  ", arg.Name, txParams[arg.Name])
}
```

I think listing the tx params in the order that they are included in tx data makes it easier to check param values included in a tx given knowledge of the contract function being called.

Reply:
good point ! fixed in 71307949